core-rust/mesh-api-server/src/mesh_api/handlers/block.rs

```diff
@@ -0,0 +1,190 @@
+        )),
+        operations,
+        related_transactions: None,
+        metadata: None,
+    };
+
+    // see https://docs.cdp.coinbase.com/mesh/docs/models#blocktransactionresponse
+    Ok(Json(models::BlockTransactionResponse {
+        transaction: Box::new(transaction),
+    }))
+}
+
+fn get_operations_and_transaction_identifier(
+    mapping_context: &MappingContext,
+    database: &StateManagerDatabase<impl ReadableRocks>,
+    state_version: StateVersion,
+    transaction_identifiers: &CommittedTransactionIdentifiers,
+    requested_transaction_identifier: Option<&str>,
+) -> Result<(Vec<models::Operation>, String), ResponseError> {
+    let (operations, transaction_identifier) = match transaction_identifiers
+        .transaction_hashes
+        .as_user()
+    {
+        // In case of non-user transaction we return empty transaction,
+        // otherwise mesh-cli returns error.
+        // Unfortunately non-user transactions don't have txid,
+        // let's use state_version as transaction_identifier.
+        None => {
+            let transaction_identifier = format!("state_version_{}", state_version);
+            if requested_transaction_identifier.is_some_and(|tx_id| tx_id != transaction_identifier)
+            {
+                return Err(ResponseError::from(ApiError::InvalidRequest)
+                    .with_details("transaction_identifier does not match with block_identifier"));
+            }
+
+            (vec![], transaction_identifier)
+        }
+        Some(user_hashes) => {
+            let local_transaction_execution = database
+                .get_committed_local_transaction_execution(state_version)
+                .ok_or_else(|| {
+                    ResponseError::from(ApiError::TransactionNotFound).with_details(format!(
+                        "Failed fetching transaction execution for state version {}",
+                        state_version.number()
+                    ))
+                })?;
+
+            let transaction_identifier =
+                to_api_hash_bech32m(mapping_context, &user_hashes.transaction_intent_hash)?;
+
+            if requested_transaction_identifier.is_some_and(|tx_id| tx_id != transaction_identifier)
+            {
+                return Err(ResponseError::from(ApiError::InvalidRequest)
+                    .with_details("transaction_identifier does not match with block_identifier"));
+            }
+
+            let status = MeshApiOperationStatus::from(local_transaction_execution.outcome);
+
+            let mut index = 0_i64;
+            let mut operations = vec![];
+            for (address, balance_changes) in local_transaction_execution
+                .global_balance_summary
+                .global_balance_changes
+            {
+                if address.is_account() {
+                    for (resource_address, balance_change) in balance_changes {
+                        if let BalanceChange::Fungible(amount) = balance_change {
```

Comment:
This is okay for a first pass, but we will want to fix this in due course to use a combination of:
* Balance changes & Fee lock data
* Mint/burn events

Reply:
Makes sense

Reply:
TODO added